### ISSUE TITLE
Disabling alias silently disables enabled linters

### ISSUE THREAD

**Issue by matthewhughes934** [golangci/golangci-lint#3746]
### Welcome

- [X] Yes, I'm using a binary release within 2 latest major releases. Only such installations are supported.
- [X] Yes, I've searched similar issues on GitHub and didn't find any.
- [X] Yes, I've included all information below (version, config, etc.).
- [X] Yes, I've tried with the standalone linter if available (e.g., gocritic, go vet, etc.). (https://golangci-lint.run/usage/linters/)

### Description of the problem

Given the file:

```go
$ cat bare.go 
package barepkg

func SomeFunc() {
	if 1 == 1 {}	// trigger staticcheck error
}
```

Running with `staticcheck` reports an issue as expected:

```console
$ ./golangci-lint run --no-config --enable staticcheck bare.go 
bare.go:4:2: SA9003: empty branch (staticcheck)
	if 1 == 1 {}	// trigger staticcheck error
	^
```

Disabling and enabling `staticcheck` errors as expected:

```console
$ ./golangci-lint run --no-config --enable staticcheck --disable staticcheck bare.go 
ERRO Running error: linter "staticcheck" can't be disabled and enabled at one moment 
```

**but** Disabling `megacheck` (alias for `staticcheck`) and enabling `staticcheck` silently disables `staticcheck`:

```console
$ ./golangci-lint run --no-config --enable staticcheck --disable megacheck bare.go
$ echo $?
0
```

### Version of golangci-lint

Tested on `1.52.1` and the current head of master: ca05239caf32d9fcda32bb0f8d53e134c290e3df

```console
$ golangci-lint --version
golangci-lint has version 1.52.1 built with go1.20.2 from d92b38cc on 2023-03-21T19:48:38Z
$ ./golangci-lint --version
golangci-lint has version (devel) built with go1.20.2 from (unknown, mod sum: "") on (unknown
```

### Configuration file

No config, see flags in description

### Go environment

<details>

```console
$ go version && go env | sed 's/mjh/me/g'
go version go1.20.2 linux/amd64
GO111MODULE=""
GOARCH="amd64"
GOBIN=""
GOCACHE="/home/me/.cache/go-build"
GOENV="/home/me/.config/go/env"
GOEXE=""
GOEXPERIMENT=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOINSECURE=""
GOMODCACHE="/home/me/.local/share/go/pkg/mod"
GONOPROXY=""
GONOSUMDB=""
GOOS="linux"
GOPATH="/home/me/.local/share/go"
GOPRIVATE=""
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/usr/lib/go"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/usr/lib/go/pkg/tool/linux_amd64"
GOVCS=""
GOVERSION="go1.20.2"
GCCGO="gccgo"
GOAMD64="v1"
AR="ar"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD="/home/me/src/golangci-lint/go.mod"
GOWORK=""
CGO_CFLAGS="-O2 -g"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-O2 -g"
CGO_FFLAGS="-O2 -g"
CGO_LDFLAGS="-O2 -g"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -Wl,--no-gc-sections -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build635768465=/tmp/go-build -gno-record-gcc-switches
```

### Verbose output of running

Adding verbosity to example above

```console
$ golangci-lint cache clean
$ ./golangci-lint run --verbose --no-config --enable staticcheck --disable megacheck bare.go 
INFO [lintersdb] Active 4 linters: [errcheck govet ineffassign typecheck] 
INFO [loader] Go packages loading at mode 575 (imports|name|types_sizes|deps|files|compiled_files|exports_file) took 12.13721ms 
INFO [runner/filename_unadjuster] Pre-built 0 adjustments in 165.104µs 
INFO [linters_context/goanalysis] analyzers took 0s with no stages 
INFO [runner] processing took 33.805µs with stages: max_same_issues: 1.956µs, path_prettifier: 1.955µs, cgo: 1.886µs, skip_dirs: 1.886µs, uniq_by_line: 1.886µs, nolint: 1.886µs, autogenerated_exclude: 1.886µs, fixer: 1.816µs, max_per_file_from_linter: 1.467µs, sort_results: 1.467µs, severity-rules: 1.467µs, exclude: 1.467µs, identifier_marker: 1.467µs, path_prefixer: 1.467µs, filename_unadjuster: 1.466µs, source_code: 1.397µs, diff: 1.397µs, exclude-rules: 1.397µs, max_from_linter: 1.397µs, path_shortener: 1.396µs, skip_files: 1.396µs 
INFO [runner] linters took 536.516µs with stages: goanalysis_metalinter: 395.718µs 
INFO File cache stats: 0 entries of total size 0B 
INFO Memory: 2 samples, avg is 31.6MB, max is 31.6MB 
INFO Execution took 18.626256ms 
```

### Code example or link to a public repository

See description

**Comment by matthewhughes934** [comment]
I had an attempt at changing the behaviour here: https://github.com/golangci/golangci-lint/compare/master...matthewhughes934:golangci-lint:fix-silent-linter-disable-through-alias

**Comment by ldez** [comment]
Hello,

It is a side effect of a "feature" because `megacheck` is a kind of set.
From what I can read in the code, the goal was to provide a compatibility layer when `megacheck` was split into 3 linters, i.e., enable/disable the 3 linters simultaneously.

```console
$ golangci-lint run -v --disable-all -Emegacheck
... 
INFO [lintersdb] Active 3 linters: [gosimple staticcheck unused]
...
```

```console
$ golangci-lint run -v -Dmegacheck
...
INFO [lintersdb] Active 4 linters: [errcheck govet ineffassign typecheck]
```

```console
$ golangci-lint run -v            
...
INFO [lintersdb] Active 7 linters: [errcheck gosimple govet ineffassign staticcheck typecheck unused]
...
```

**Comment by ldez** [comment]
To be precise I think I was expected: as `megacheck` is a kind of set, the possibility to enable/disable one of the linter is expected.

the following test is explaining the expected behavior:
https://github.com/golangci/golangci-lint/blob/ca05239caf32d9fcda32bb0f8d53e134c290e3df/pkg/lint/lintersdb/enabled_set_test.go#L14

**Comment by ldez** [comment]
For now, this issue cannot be fixed.

**Comment by matthewhughes934** [comment]
Thanks for the response, and thanks for the extra info :slightly_smiling_face:

**Comment by ldez** [comment]
now alternative names are deprecated, so we can consider this issue as fixed.
